lib/client-assets.php

```diff
@@ -243,8 +243,8 @@ function gutenberg_register_packages_scripts( $scripts ) {
 		// For example, `…/build/a11y/index.min.js` becomes `wp-a11y`.
 		$handle = 'wp-' . basename( dirname( $path ) );
 
-		// Replace `.js` extension with `.asset.php` to find the generated dependencies file.
-		$asset_file   = substr( $path, 0, -3 ) . '.asset.php';
+		// Replace suffix and extension with `.asset.php` to find the generated dependencies file.
+		$asset_file   = substr( $path, 0, -( strlen( $suffix ) ) ) . '.asset.php';
```

Comment:
We could hardcode `index.asset.php` here, but I found it required less changes this way.

Reply:
There are asset files that are not `index`, e.g., `frontend.asset.php` for blocks. These are not used in this particular loop but who knows, someday they could. Being flexible sounds OK.